### PR TITLE
MAINT: fix `.gitignore` issues, and remove remaining `NPY_INLINE` usages

### PR DESCRIPTION
**Pull request by rgommers**
The source files being listed in `.gitignore` caused ripgrep to ignore them, it probably has other undesirable side-effects. This came in when converting templated files to plain C files.

This is a follow-up to gh-22674.

For future reference, here is how to check if files in `.gitignore` are tracked:
```python
import os

with open('../numpy/.gitignore') as f:
    ignored = f.readlines()
    ignored = [s.split('\n')[0] for s in ignored if s.startswith('numpy')]

for fname in ignored:
    if os.path.exists(fname):
        print(fname)
 ```

### CODE REVIEW

File: numpy/core/src/common/npy_cpu_features.c [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #include "npy_cpu_features.h"
 #include "npy_cpu_dispatch.h" // To guarantee the CPU baseline definitions are in scope.
-#include "numpy/npy_common.h" // for NPY_INLINE
```
Comment (seberg) [inline numpy/core/src/common/npy_cpu_features.c:3]: Seems the include was necessary also for `ndarraytypes.h` indirectly.  But then I guess you can also just replace those types instead.  They are around line 454 (only active on x86 apple).

Comment (rgommers) [reply numpy/core/src/common/npy_cpu_features.c:3]: Using `ndarraytypes.h` yielded a complaint about using deprecated API ([CI log](https://github.com/numpy/numpy/actions/runs/3558493802/jobs/5977394229)), so I just put `npy_common.h` back.

**Comment by rgommers** [comment]
This should be good to merge now.

**Review by mattip — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @rgommers

**Pull request merged**